text/0000-components.md

```diff
@@ -0,0 +1,261 @@
+SELECT FROM counts_table name,count WHERE start <= 3 AND 3 < end;
+```
+
+![Example Query Output](../images/grid_components_state_delta_4.svg)
+
+Using this model, records may be joined across tables at a given block height.
+This provides consistent views for the data at any point in time.  For example,
+if a block is committed while paging records, the resulting count of records at
+the initial block height queried will remain the same.
+
+## Command Line Tool (grid)
+
+The Grid CLI (`grid`) provides an command-line interface for interacting with the
+Grid Daemon’s REST API, with the Grid CLI mirroring the functionality available
+in the REST API. For example, the Grid CLI will provide functionality to:
+
+- Submit transactions
+- Read/query schema data
+- Read/query Pike identity data
+- Read/query product data
+- Read/query track and trace data
+
+The CLI will also contain administrative commands, such as initializing
+a PostgreSQL instance with the necessary tables for State Delta Export.
+
+The Grid CLI should feel natural on a Linux system and adhere to Linux CLI
+norms and should be suitable for use when scripting. For example, when an error
+occurs, the CLI command should exit with a non-zero exit status so that it can
+be checked in a bash script. The command’s output should be both human friendly
+and easily parsable with common Linux command line tools such as awk and sed
+(when necessary, as different command-line options). 
+
+The Grid CLI will use the Rust clap library to parse command line arguments,
+with restrictions. Problematic multiple-value syntax such as `-o <val1> <val2>`
+should be avoided and `-o <val1> -o <val2>` form used instead.
+
+## SDK
+
+The Grid SDK provides programming support for application developers and
+systems integrators. Initially, only the Rust programming language is
+supported, though others may be proposed in future RFCs.
+
+In particular, the SDK includes APIs and implementations to help write smart
+contracts, including:
+
+- Reading product information from state
+- Reading Pike identity information from state
+- Reading schema information from state
+- Pike permission verifier
+- Schema compliance verify
+
+Like the other components, the SDK will grow over time as new features are
+supported by Grid. When adding new features or making changes to the SDK,
+backwards compatibility should be considered. The SDK’s versioning should
+conform to semver (which is true of all components, but of particular
+importance for the SDK).
+Smart Contracts
+
+Several features of Hyperledger Grid use smart contracts as part of their
+implementation. The smart contracts are deployed in Sabre, which is part of
+Hyperledger Sawtooth. Sabre executes smart contracts which have been compiled
+to WebAssembly. Most of the smart contracts which are formally part of
+Hyperledger Grid will be written in the Rust programming language. However,
+future RFCs may extend this to additional programming languages supported by
```

Comment:
s/extend this to/provide SDKs for/

...because the referent of "this" is unclear (grammatically speaking).